miio/gateway.py

```diff
@@ -646,71 +646,112 @@ class GatewayLight(GatewayDevice):
     """Light controls for the gateway."""
 
     @command()
-    def get_night_light_rgb(self):
-        """Unknown."""
-        # Returns 0 when light is off?"""
-        # looks like this is the same as get_rgb
-        # id': 65064, 'method': 'set_night_light_rgb', 'params': [419407616]}
-        # {'method': 'props', 'params':
-        # {'light': 'on', 'from.light': '4,,,'}, 'id': 88457} ?!
-        return self._gateway.send("get_night_light_rgb")
+    def rgb_status(self):
+        """Get current status of the light (always correct)."""
+        # Returns {"is_on": false, "brightness": 0, "rgb": (0, 0, 0)} when light is off
+        state_int = self._gateway.send("get_rgb").pop()
+        brightness = int_to_brightness(state_int)
+        rgb = int_to_rgb(state_int)
+        if brightness > 0:
+            is_on = True
+        else:
+            is_on = False
+        return {"is_on": is_on, "brightness": brightness, "rgb": rgb}
+
+    @command()
+    def night_light_status(self):
+        """Get status of the night light."""
+        # The night light controls the same LEDs as the rgb light, but has a seprate memory for brightness and color.
+        # Changing the rgb light does not effect the stored state of the night light, while changing the night light does effect the state of the rgb light.
+        # Therefore this command only gives the correct status of the LEDs if the last command was a night light command and not a rgb light command, otherwise it gives the stored values of the night light.
+        # Returns {"is_on": false, "brightness": 0, "rgb": (0, 0, 0)} when night light is off
+        state_int = self._gateway.send("get_night_light_rgb").pop()
+        brightness = int_to_brightness(state_int)
+        rgb = int_to_rgb(state_int)
+        if brightness > 0:
+            is_on = True
+        else:
+            is_on = False
+        return {"is_on": is_on, "brightness": brightness, "rgb": rgb}
+
+    @command(
+        click.argument("brightness", type=int),
+        click.argument("rgb", type=(int, int, int)),
+    )
+    def set_rgb(self, brightness, rgb):
```

Comment:
```suggestion
    def set_rgb(self, brightness: int, rgb: Tuple[int, int, int]):
```

Reply:
Type annotations should also be added to other other functions.

Reply:
thanks, did it for all functions of this PR